sa/sa.go

```diff
@@ -2216,3 +2216,70 @@ func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentR
 	}
 	return nil
 }
+
+// GetRevokedCerts gets a request specifying an issuer and a period of time,
+// and writes to the output stream the set of all certificates issued by that
+// issuer which expire during that period of time and which have been revoked.
+func (ssa *SQLStorageAuthority) GetRevokedCerts(req *sapb.GetRevokedCertsRequest, stream sapb.StorageAuthority_GetRevokedCertsServer) error {
+	atTime := time.Unix(0, req.RevokedBefore)
+
+	query := `SELECT serial, status, revokedReason, revokedDate
+		FROM certificateStatus
+		WHERE notAfter >= ?
+		AND notAfter < ?
+		AND issuerID = ?
+		AND status = ?`
+	params := []interface{}{
+		time.Unix(0, req.ExpiresAfter),
+		time.Unix(0, req.ExpiresBefore),
+		req.IssuerNameID,
+		core.OCSPStatusRevoked,
+	}
+
+	rows, err := ssa.dbReadOnlyMap.WithContext(stream.Context()).Query(query, params...)
+	if err != nil {
+		return fmt.Errorf("failed to read db: %w", err)
```

Comment:
Style-wise, I'm trending away from "failed to X: " for error messages in preference for "Xing: ". The fact that you're in an error message means failure is implicit, plus Xing is a little shorter.

```suggestion
		return fmt.Errorf("reading db: %w", err)
```

However, in this case the error will actually come from WrappedMap, which already has the responsibility of making more informative errors. It should generate an error like `failed to select certificateStatus: ...`. Except... WrappedMap doesn't actually intercept `Query`. Want to add that? Then this can just return `err`.